doc/api/vm.md

```diff
@@ -310,6 +310,27 @@ console.log(util.inspect(sandbox));
 // { globalVar: 1024 }
 ```
 
+## vm.moveMessagePortToContext(port, contextifiedSandbox)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `port` {MessagePort}
+* `contextifiedSandbox` {Object} A contextified object as returned by the
+  `vm.createContext()` method.
+* Returns: {MessagePort}
+
+Bind a `MessagePort` to a specific VM context. This returns a new `MessagePort`
+object, whose prototype and methods act as if they were created in the passed
+context. The received messages will also be emitted as objects from the passed
+context.
+
+Note that the return instance is *not* an `EventEmitter`; for receiving
+messages, the `.onmessage` property can be used.
```

Comment:
It seems rather crude to have some MessagePorts being EventEmitters, and some not. I understand you are still working on this part, and that the EventEmitter is implemented in JS complicates things. But I'd rather have it all one way or the other.

Reply:
> But I'd rather have it all one way or the other.

Me too. :) But like with the `Buffer`s, it’s just a problem that not all contexts have a concept of `EventEmitter`s … this would require making some internal modules multi-context ready, which I would like to consider out of scope for this PR. :) Also, this is already quite an advanced feature on its own.